tests/benchmarks/sched/README.rst

```diff
@@ -0,0 +1,31 @@
+Scheduler Microbenchmark
+########################
+
+This is a scheduler microbenchmark, designed to measure minimum
+latencies (not scaling performance) of specific low level scheduling
+primitives independent of overhead from application or API
+abstractions.  It works very simply: a main thread creates a "partner"
+thread at a higher priority, the partner then sleeps using
+_pend_current_thread().  From this initial state:
+
+1. The main thread calls _unpend_first_thread()
+2. The main thread calls _ready_thread()
+3. The main thread calls k_yield()
+   (the kernel switches to the partner thread)
+4. The partner thread then runs and calls _pend_current_thread() again
+   (the kernel switches to the main thread)
+5. The main thread returns from k_yield()
+
+It then iterates this many times, reporting timestamp latencies
+between each numbered step and for the whole cycle, and a running
+average for all cycles run.
+
+Note that because this involves no timer interaction (except, on some
+architectures, k_cycle_get_32()), it works correctly when run in qemu
+using the -icount argument, which can produce 100% deterministic
+behavior (not cycle-exact hardware simulation, but exactly N
+instructions per simulated nanosecond).  You can enable this using an
+environment variable set at cmake time (it's not enough to do this for
```

Comment:
I think you're saying this environment variable needs to be set for the build and running of the test, right?  Maybe change this to a positive statement saying this.

Reply:
No, I'm warning that the environment variable needs to be set for the first stage (cmake) of building the test, not just the second one (make/ninja) where environment variables are commonly interpreted.   Clarified, hopefully.